ZuluContent/Zulu/Spells/Earth/ShapeshiftSpell.cs

```diff
@@ -1,251 +1,116 @@
 using System;
 using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Scripts.Zulu.Engines.Classes;
+using Scripts.Zulu.Utilities;
 using Server;
 using Server.Network;
 using Server.Items;
 using Server.Gumps;
 using Server.Mobiles;
 using Server.Spells;
 using Server.Targeting;
+using ZuluContent.Zulu.Engines.Magic.Enchantments.Buffs;
 
 namespace Scripts.Zulu.Spells.Earth
 {
-    // polymorph into a variety of animals, one of which is the little bird
-    public class ShapeshiftSpell : AbstractEarthSpell
+    public class ShapeshiftSpell : EarthSpell, IAsyncSpell
     {
-        private Entry[] m_Entries =
+        public override TimeSpan CastDelayBase => TimeSpan.FromSeconds(0);
+    
+        public override double RequiredSkill => 60.0;
+    
+        public override int RequiredMana => 5;
+        
+        private record ShapeshiftData
         {
-            new Entry("a bird", 0x6, 0x20ee, 10),
-            new Entry("an eagle", 0x5, 0x211d, 20),
-            new Entry("a dog", 0xd9, 0x211c, 30),
-            new Entry("a wolf", 0xe1, 0x20ea, 40),
-            new Entry("a deer", 0xed, 0x20d4, 45),
-            new Entry("a panther", 0xd6, 0x2119, 50),
-            //new Entry("a bear", 0xd3, 0x2118, 50),
-            new Entry("a grizzly", 0xd4, 0x211e, 60),
-            new Entry("a polar bear", 0xd5, 0x20e1, 65),
-            new Entry("a giant serpent", 0x15, 0x20fe, 70),
-            new Entry("an earth elemental", 0xe, 0x20d7, 80),
-            new Entry("a fire elemental", 0xf, 0x20f3, 90),
-            new Entry("a water elemental", 0x10, 0x210b, 95),
-            new Entry("an air elemental", 0xd, 0x20ed, 97),
-            new Entry("a dragon", 0x3b, 0x20d6, 100),
-            new Entry("a reaper", 0x2f, 0x20fa, 110),
-            new Entry("a wisp", 0x3a, 0x2100, 120)
-        };
-
-        public class Entry
-        {
-            public readonly int BodyId;
-            public readonly int ArtId;
-            public readonly double Difficulty;
-            public readonly string Description;
-
-            public Entry(string desc, int artId, int bodyId, double difficulty)
-            {
-                BodyId = bodyId;
-                ArtId = artId;
-                Description = desc;
-                Difficulty = difficulty;
-            }
-        }
-
-
-        public override TimeSpan CastDelayBase
-        {
-            get { return TimeSpan.FromSeconds(3); }
+            public int Difficulty { get; init; }
+            public int Points { get; init; }
         }
 
-        public override double RequiredSkill
-        {
-            get
+        private Dictionary<int, ShapeshiftData> BodyData { get; set; } =
```

Comment:
This can be `private static readonly`. Also for simple types like ShapeshiftData I'd be tempted to just use a Tuple, e.g.

```csharp
private static readonly Dictionary<int, (int Difficulty, int Points)> BodyData = new()
{
    { PolymorphEntry.Bird.BodyId, (Difficulty: 10, Points: 100) },
    // or short syntax
    { PolymorphEntry.Eagle.BodyId, (20, 200) }
};
```